src/components/AttachmentsModal/AttachmentsModal.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react'
+import { Modal, Segment, Button, Icon } from 'semantic-ui-react'
+import { Component } from 'react'
+
+class AttachmentsModal extends Component {
+    state = { open: false }
+  
+    open = () => this.setState({ open: true })
```

Comment:
Use the class style like real function. Do not forget to bind this function with this in the constructor.